### ISSUE TITLE
Pragma checking does support complex cases (semver) like >=, ^, <, ==

### ISSUE THREAD

**Issue by arjunaskykok** [vyperlang/vyper#1993]
## Simple Summary
Currently we can add pragma in Vyper file, for example:

`# @version 0.1.0b13`

It means Vyper compiler version 0.2.0 will fail to compile this Vyper file. But this is too primitive. We need to add [semver](https://docs.npmjs.com/misc/semver) capability.  It needs to support >=, ^, <, ==, for example:

`# @version ==0.1.0b13`

This pragma line means only Vyper compiler version 0.1.0b13 can compile this Vyper file successfully.

## Motivation
Solidity pragma supports semver capability.

## Specification
Semver specification is here: https://docs.npmjs.com/misc/semver

## Backwards Compatibility

It's safe.

## References

https://github.com/python-semver/python-semver
https://vyper.readthedocs.io/en/latest/structure-of-a-contract.html
https://solidity.readthedocs.io/en/v0.6.8/layout-of-source-files.html

## Copyright
Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/)

**Comment by arjunaskykok** [comment]
Is it okay to use third party package (python-semver) to implement this feature?

**Comment by fubuloubu** [comment]
Yeah, I think the `semver` package has no external dependencies, so it's safe to use

**Comment by fubuloubu** [comment]
https://github.com/rbarrois/python-semanticversion

**Comment by fubuloubu** [comment]
Note: Move to contract-level natspec

**Comment by fubuloubu** [comment]
Note: Mostly fixed in https://github.com/vyperlang/vyper/pull/1995

Still need to discuss move to contract-level natpsec for easier parsing and sharing
